### ISSUE TITLE
[feature] Hot Edges (with mouse click activation)

### ISSUE THREAD

**Issue by Solarunit** [solus-project/budgie-desktop#606]
I think it would be awesome!

Some examples of implementation:

- Right mouse click at the Right Edge of the screen opens applets in Raven
- Middle mouse click at the Right Edge of the screen opens notifications in Raven 
- Scroll wheel: workspaces overview
- many others (e.g. custom menu in raven with user pinned applications)

![Alt Text](http://i.imgur.com/5Hw4qzc.gif)

Advantages:
+ It's super fast, notifications/applets is only one click away.
+ No accidental activation which is common in "hot corners" tools and also no annoying delay.

**Comment by Snisser** [comment]
Isn't that more of a **pull request** and not an **issue**? :P

**Comment by ikeydoherty** [comment]
What happens when you have multiple monitors.. ?

**Comment by Snisser** [comment]
Good Idea but Ikey's point is legit. Raven can only be shown on the **Main Monitor's right side**, when a secondary monitor is on the right of it, you wouldn't be able to switch monitors with the cursor without triggering Raven, kind of annoying, isn't it? :P

**Comment by Solarunit** [comment]
> What happens when you have multiple monitors.. ?

Well, it is obvious that we can't use neighboring edges. BTW The same problem is also applies to the Hot corners tools. In any case, we have 4 (6) edges we can use:

![Alt Text](http://i.imgur.com/pL76ymK.png)

> Good Idea but Ikey's point is legit. Raven can only be shown on the main monitor's right side. If a secondary monitor is on the right of it, you wouldn't be able to switch monitors with the cursor without triggering Raven, kind of annoying, isn't it? :P

No, of course you can switch monitors with the cursor because hot edge is activated **with mouse click**  
 ------------------

PS: Raven activation is only one example. We can use hot edges for virtual desktop switching.  
Middle mouse click at the Right Edge of the screen:

![Alt Text](http://i.imgur.com/MmiQcUO.gif)

**Comment by snkiz** [comment]
you can make mulit-monitor setups that do fancy things via click and/or pressure reveal with compiz. Click for hot edge push to switch monitors/desktops. It was handy when I used it, very fast for desktop/app management using scale and expo. Its not essential, but really would be nice for converts used to this behaviour. I often forget and click the screen edges in budgie, nothing happens.

**Comment by DasJott** [comment]
Such an overview of the desktops where one can drag and drop windows between would be a nice thing!
